File: node/7/MyLanguages/test_my_languages.js
const assert = require("assert");
var func = require("./my_languages");

let ptr = [
    [{"Java" : 10, "Ruby" : 80, "Python" : 65}, ["Ruby", "Python"]],
    [{"Hindi" : 60, "Greek" : 71, "Dutch" : 93}, ["Dutch", "Greek", "Hindi"]],
    [{"C++" : 50, "ASM" : 10, "Haskell" : 20}, []],
];

ptr.forEach(function(p) {
    var inp = p[0];
    var exp = p[1];

    // [1, 2, 3] === [1, 2, 3] is false with strictEqual()
    assert.ok(func(inp), exp);
});

File: python/7/MyLanguages/test_my_languages.py
from unittest import TestCase
from unittest import main

from .my_languages import my_languages


class TestMyLanguages(TestCase):
    def test_my_languages(self):
        ptr = [
            ({'Java': 10, 'Ruby': 80, 'Python': 65}, ['Ruby', 'Python']),
            ({'Hindi': 60, 'Dutch': 93, 'Greek': 71}, ['Dutch', 'Greek', 'Hindi']),
            ({'C++': 50, 'ASM': 10, 'Haskell': 20}, []),
        ]
        for inp, exp in ptr:
            with self.subTest(inp=inp, exp=exp):
                self.assertEqual(my_languages(results=inp), exp)


if __name__ == '__main__':
    main(verbosity=2)
